### DISCUSSION TITLE
[Feature request] Gamification

### DISCUSSION THREAD

**Discussion by delta0815** [donetick/donetick#23] [Ideas]
Are there any plans to add some gamification?

I think of two ideas:

1. Every tasks has some credits assigned. Whoever does the task gets those credits and after a certain amount of credits your receive badges.
2. Currently tasks can be accomplished by either manually marking them as done or scanning an assigned NFC tag. It would be cool if the admin could make some tasks only completable by NFC scanning. With this you could force your family members to get to a specific location in order to achieve a task (like bringing out the trash and scan the NFC tag which sticks at the dumpster.

Curious what you guys think about it :-)

Cheers,

quicky

**Comment by ados8** [comment]
I can definitely see gamification helping to encourage kids to complete their tasks.
I like both ideas, but with option 1 there should be an option that the admin has to approve the credits.
This would eliminate the kids completing tasks to low standards in an effort to quickly get credits. 😊
Building on that, credits should stay in a person's account like a bank and the admin can deduct points from the account.
Effectively the process would be:

1. Child has list of tasks for the week
2. Child completes tasks and admin gets notified
3. Admin checks and approves the credits (or says do it properly 😋)
4. Child asks to play video games, admin checks and deducts points from the account for fun time

**Reply by FracKenA** [reply]
i would like to see this, i am looking for a solution to replace famzoo that lets me assign a currency value to chores. this is how my kids get their allowance. how can i assist with this? i don't want to just cut loose doing a pr that won't be accepted.

**Comment by Sheephog** [comment]
Just some thoughts on this subject. With points already starting to be implemented, i think a good idea is the "admin area" where points are able to be reviewed/deducted. Maybe admin could be able to create rewards/vouchers that are able to be 'bought' by the user. i.e "double dessert - 100 points" and then the admin can then see a list of "vouchers" and mark them as used/redeemed as and when necessary.  The admin area can also be expanded to include sections for creating accounts/resetting passwords, limiting further signups, adjusting other account permissions, and more. 
+1 for admin approval for task completion also (this should be an option on a per account basis, adjustable in the aforementioned 'admin area' ) 
Just some food for thought anyway, plenty of ideas coming your way I'm sure. Only so much time to develop.

**Reply by ados8** [reply]
Admin portal was requested [here](https://github.com/donetick/donetick/discussions/50).
I'm not sure if it will be implemented.

**Reply by meauxt** [reply]
To keep things tidy, I'm thinking of adding a note section to the redeem section instead of maintaining a list of rewards.

**Reply by delta0815** [reply]
I agree, I personally think the motivation comes from the challenge itself. I'd rather like to see a weekly summary (in general) which shows who owned the most points. But even the points itself as they are included by now it's already a great improvement.

**Reply by ledfloyd1984** [reply]
I think just having a store area where you could spend your points would be nice. That would make it easier from the dev perspective, that way there isn't a bunch of refactoring the existing admin stuff. The circle admin could be the one to add items to the store. Or if you could simply use your accumulated points to defer a task that is assigned to you to another person.

**Reply by ados8** [reply]
Any updates on this one.
I don't think there's any need to maintain a list, have that list be user controlled.
Right now I'm using HabbitTrove but would like to use one app.
The system there is a page dedicated for what you can spend the points on.
The admin could make the list to their liking within the circle.

**Comment by ados8** [comment]
For those wanting gamification I recently discovered Habitica.
It's not as powerful from a task perspective but has an extensive points system.
In the end I had issues with MongoDB, always have with that database system, so I trying out HabitTrove.
No "game" elements but it has a points system which is being developed out and a modern interface.
This isn't a replacement to Donetick in any regard but you can run additionally.

**Comment by hales8181** [comment]
I like the idea of a weekly scoreboard, I have 3 kids and I can imagine this would be a great motivator. 
+1 also for the having to approve that the chore has actually been done.

**Comment by cbrinicombe13** [comment]
Could have a system whereby:
- You track the number of tasks completed by everyone
- You build achievements at 10,15,20 tasks etc. Each one can award a badge.
- You track the number of consecutive days/weeks people complete a task and show this as a 'streak' (Sometimes kind of annoying, but i like them...)

Might be interesting and something like [trophy](https://www.trophy.so/) can do a lot of the legwork.

**Comment by JohnFawkes** [comment]
personally i use stuff like this for chores for my kids. Im personally using habitrove as it has a wishlist that i can add things to for my kids to redeem their point. if donetick could add something like this and a simple way to upload a picture to validate the task was done and recive a discord notification when an account completes the task, I'd be full on board with donetick

**Comment by brendonmconnett** [comment]
Any update on this user's comment about a "store" where users can redeem certain rewards set by the admin? This would be really cool because it creates a more automated flow versus having to manually deduct points from the user's account. https://github.com/donetick/donetick/discussions/23#discussioncomment-12391434
